### DISCUSSION TITLE
Vuepress PWA 2.0.0-rc.17 issue

### DISCUSSION THREAD

**Discussion by libellux** [vuepress/ecosystem#80] [Q&A] [answered]
hello,

since I've updated to the latest version of both vuepress and the pwa plugin I receive a build issue.

```
  "devDependencies": {
    "@vuepress/bundler-vite": "2.0.0-rc.8",
    "@vuepress/plugin-docsearch": "2.0.0-rc.17",
    "@vuepress/plugin-google-analytics": "2.0.0-rc.15",
    "@vuepress/plugin-pwa": "2.0.0-rc.17",
    "@vuepress/plugin-sitemap": "2.0.0-rc.17",
    "@vuepress/theme-default": "2.0.0-rc.17",
    "vue": "^3.4.20",
    "vuepress": "2.0.0-rc.8"
  }
```

and the vuepress config file would be:

```
import { pwaPlugin } from '@vuepress/plugin-pwa'

export default defineUserConfig({
  shouldPrefetch: false,
  plugins: [
    pwaPlugin({
      skipWaiting: true
    }),
```

However, whenever I try to build vuepress with the PWA plugin I receive the following error:

```
TypeError: Cannot read properties of undefined (reading '0')
    at file:///.../node_modules/.pnpm/@vuepress+plugin-pwa@2.0.0-rc.17_vuepress@2.0.0-rc.8/node_modules/@vuepress/plugin-pwa/lib/node/injectLinksToHead.js:7:17
    at Array.forEach (<anonymous>)
```

I'm not sure on how to approach this as it seems the service worker is somewhat broken as I cannot go back using prior versions either. which worked before I bumped vuepress + pwa versions.

Any help or if ideas to push me in the right direction would be appreciated 👍 

Best regards

**Comment by Mister-Hope** [comment, marked as answer]
### A full reproduction is required, check if you have invalid head option in config file.

**Reply by libellux** [reply]
Thanks! was indeed misconfigured head data.
